GliaWidgets/ViewController/Survey/SurveyViewController.View.swift

```diff
@@ -0,0 +1,116 @@
+            $0.backgroundColor = .white
+            $0.layer.masksToBounds = false
+            $0.layer.shadowColor = UIColor.black.cgColor
+            $0.layer.shadowOpacity = 0.15
+            $0.layer.shadowOffset = .zero
+            $0.layer.shadowRadius = 8
+
+        }
+        let cancelButton = UIButton(type: .custom).make {
+            $0.setTitle("Cancel", for: .normal)
+            $0.backgroundColor = .init(hex: "#D11149")
+            $0.titleLabel?.font = .systemFont(ofSize: 16)
+            $0.layer.cornerRadius = 4
+        }
+        let submitButton = UIButton(type: .custom).make {
+            $0.setTitle("Submit", for: .normal)
+            $0.backgroundColor = .init(hex: "#0F6BFF")
+            $0.titleLabel?.font = .systemFont(ofSize: 16)
+            $0.layer.cornerRadius = 4
+        }
+        lazy var buttonStackView = UIStackView.make(.horizontal, spacing: 16)(
+            cancelButton,
+            submitButton
+        ).makeView()
+
+        override func setup() {
+            super.setup()
+
+            backgroundColor = .white
+
+            addSubview(scrollView)
+            scrollView.addSubview(contentContainerStackView)
+
+            buttonContainer.addSubview(buttonStackView)
+            addSubview(buttonContainer)
+            addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(hideKeyboard)))
+        }
+
+        @objc
+        private func hideKeyboard(gesture: UITapGestureRecognizer) {
+            endEditing(true)
+        }
+
+        override func defineLayout() {
+            super.defineLayout()
+            var constraints = [NSLayoutConstraint](); defer { constraints.activate() }
+
+            constraints += scrollView.topAnchor.constraint(equalTo: topAnchor)
+            constraints += scrollView.leadingAnchor.constraint(equalTo: leadingAnchor)
+            constraints += scrollView.trailingAnchor.constraint(equalTo: trailingAnchor)
+            constraints += scrollView.bottomAnchor.constraint(equalTo: buttonContainer.topAnchor)
+
+            let contentPadding: CGFloat = 24
+            constraints += contentContainerStackView.topAnchor.constraint(
+                equalTo: scrollView.topAnchor,
+                constant: contentPadding
+            )
+            constraints += contentContainerStackView.leadingAnchor.constraint(
+                equalTo: scrollView.safeAreaLayoutGuide.leadingAnchor,
+                constant: contentPadding
+            )
+            constraints += contentContainerStackView.widthAnchor.constraint(
+                equalTo: scrollView.safeAreaLayoutGuide.widthAnchor,
+                constant: -2 * contentPadding
+            )
+            constraints += contentContainerStackView.bottomAnchor.constraint(
+                equalTo: scrollView.bottomAnchor,
+                constant: -contentPadding
+            )
+
+            constraints += buttonContainer.bottomAnchor.constraint(equalTo: bottomAnchor)
```

Comment:
This doesn't respect safe area

Reply:
good catch 5c5774c9127725a4a9612d39b8e585b3a081d9bc